### PR TITLE
added transitions for popup

### PR DESCRIPTION
**Pull request by eclewlow**
- Integrated Semantic UI Transitions for Popup

Inherits `animation` and `duration` prop for Transition element.

Control like this:

```javascript
        <DateInput
          ...
          animation='fly left'
          duration={300}
          ...
        />
```

### CODE REVIEW

**Comment by arfedulov** [comment]
Hi! That's cool. Thanks for contribution. But there are some problems:

* You've made a lot of code-style changes (try to run `npm run lint`)
* calling `setTimeout` on line 155 in `InputView.tsx` couses typescript error (explicitely calling `window.setTimeout` fixes that)

If you are eventually going to make some style changes please put that changes in separate commit.

**Comment by eclewlow** [comment]
Ok.  I made the changes.

**Comment by arfedulov** [comment]
### keyboard navigation isn't working

* press `tab` until some of the popups is open
* try to select some date using keyboard arrow buttons (doesn't work)

### runtime errors

* press `tab` until some of the inline calendars is in focus (runtime error occures when inline calendar gets focus)

**Comment by eclewlow** [comment]
Ok. Fixed issues.

**Comment by arfedulov** [comment]
Everything is fine. But I'm not very exited about `hideOnScroll` property is gone from Popup. It was hardcoded and I was considering make it configurable. But now if we set `hideOnScroll` on Popup it throws an error.

And the last problem is (just a question actually) what is the purpose of `isAnimating` state field on InputView? I found that deleting these two lines (if ``closable={true}``)

```jsx
        onStart={() => this.setState({ isAnimating: true })}
        onComplete={() => this.setState({ isAnimating: false })}
```
prevents popup from closing. But these would work perfectly:

```jsx
        onStart={() => this.setState({})}
        onComplete={() => this.setState({})}
```
So I'm a little bit confused about what is happening.

File: src/views/InputView.tsx [hunk 1]
```diff
@@ -232,6 +248,19 @@ class InputView extends React.Component<InputViewProps, any> {
...
+    const { closePopup } = this.props;
+    closePopup();
+  }
+
+  private setScrollListener() {
+    window.addEventListener('scroll', this.scrollListener.bind(this));
+  }
+
+  private unsetScrollListener() {
+    window.removeEventListener('scroll', this.scrollListener.bind(this));
```
Comment (arfedulov) [inline src/views/InputView.tsx]: Here you actually do not remove 'scroll' listener because `this.scrollListener.bind` creates new function instance.

**Pull request merged**